conda-store-server/conda_store_server/server/views/api.py

```diff
@@ -310,17 +310,14 @@ async def api_create_namespace(
         return {"status": "ok"}
 
 
-@router_api.put(
-    "/namespace/{namespace}/",
-    response_model=schema.APIAckResponse,
-)
-async def api_update_namespace(
+def _api_namespace_common(
+    *,
     namespace: str,
     request: Request,
-    metadata: Dict[str, Any] = None,
-    role_mappings: Dict[str, List[str]] = None,
-    auth=Depends(dependencies.get_auth),
-    conda_store=Depends(dependencies.get_conda_store),
+    auth,
+    conda_store,
+    func,
+    **kwargs,
```

Comment:
All these routes are pretty similar, so I define a common function to make it easier to write and review.